WordPress/src/main/java/org/wordpress/android/ui/stockmedia/StockMediaPickerActivity.java

```diff
@@ -0,0 +1,490 @@
+                    if (!mViewModel.isFetching()) {
+                        showProgress(false);
+                    }
+                    //noinspection ConstantConditions
+                    showEmptyView(mediaList.isEmpty()
+                                  && !TextUtils.isEmpty(mViewModel.getSearchQuery()));
+                    mAdapter.setMediaList(mediaList);
+                }
+            }
+        });
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(final MenuItem item) {
+        if (item.getItemId() == android.R.id.home) {
+            setResult(RESULT_CANCELED);
+            finish();
+            return true;
+        }
+        return super.onOptionsItemSelected(item);
+    }
+
+    @Override
+    public boolean onQueryTextSubmit(String query) {
+        if (mSearchView != null) {
+            mSearchView.clearFocus();
+        }
+        ActivityUtils.hideKeyboard(this);
+        return true;
+    }
+
+    @Override
+    public boolean onQueryTextChange(String query) {
+        submitSearch(query, true);
+        return true;
+    }
+
+    private void configureSearchView() {
+        mSearchView = findViewById(R.id.search_view);
+        mSearchView.setQuery(mViewModel.getSearchQuery(), false);
+
+        // don't allow the SearchView to be closed
+        mSearchView.setOnCloseListener(new SearchView.OnCloseListener() {
+            @Override public boolean onClose() {
+                return true;
+            }
+        });
+
+        mSearchView.setOnQueryTextListener(this);
+    }
+
+    private void showEmptyView(boolean show) {
+        if (!isFinishing()) {
+            TextView txtEmpty = findViewById(R.id.text_empty);
+            txtEmpty.setVisibility(show ? View.VISIBLE : View.GONE);
+            if (show) {
+                boolean isEmpty = mSearchQuery == null || mSearchQuery.length() < MIN_SEARCH_QUERY_SIZE;
+                if (isEmpty) {
+                    String message = getString(R.string.stock_media_picker_initial_empty_text);
+                    String subMessage = getString(R.string.stock_media_picker_initial_empty_subtext);
+                    String link = "<a href='https://pexels.com/'>Pexels</a>";
+                    String html = message
+                                  + "<br /><br />"
+                                  + "<small>" + String.format(subMessage, link) + "</small>";
+                    txtEmpty.setMovementMethod(WPLinkMovementMethod.getInstance());
+                    txtEmpty.setText(Html.fromHtml(html));
+                } else {
+                    txtEmpty.setText(R.string.stock_media_picker_empty_results);
+                }
+            }
+        }
+    }
+
+    private void showProgress(boolean show) {
+        if (!isFinishing()) {
+            findViewById(R.id.progress).setVisibility(show ? View.VISIBLE : View.GONE);
+        }
+    }
+
+    private void submitSearch(@Nullable final String query, boolean delayed) {
```

Comment:
Since this is not really activity related, I think we should have this in the `ViewModel`. We shouldn't really need to clear the adapter and manually handle the empty view. This creates an inconsistency between the view and view model. If the search will go through, we should remove the current data in the view model and post an update and let the observer pattern do its thing. I've already made a suggestion about how to handle the empty view previously. There is an example of this which starts [here](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/viewmodel/PluginBrowserViewModel.java#L454).